iiod/serial.c

```diff
@@ -258,7 +258,13 @@ int start_serial_daemon(struct iio_context *ctx, const char *uart_params,
 	IIO_DEBUG("Serving over UART on %s at %u bps, %u bits\n",
 		  dev, uart_bps, uart_bits);
 
-	return thread_pool_add_thread(pool, serial_main, pdata, "iiod_serial_thd");
+	err = thread_pool_add_thread(pool, serial_main, pdata, "iiod_serial_thd");
+	if (err)
+		goto err_close_fd;
+
+	free(dev);
+
+	return 0;
```

Comment:
Hmm, maybe there's more to it (or at least to be said in the commit description). This change seems to be more than just `free(dev);`. Now, in case `thread_pool_add_thread()` fails, we are also cleaning pdata and fd. Before we were just directly returning `thread_pool_add_thread()`. This came to my attention because I was thinking that we could do:

```
...
free(dev);
return thread_pool_add_thread();
...
```

Am I missing something?

Reply:
@nunojsa If `thread_pool_add_thread` fails then we do want to clean up, yes. It should have been done before. You are right that it should be described in the commit message. I will just split it into a separate commit.

Reply:
@nunojsa fixed now, thanks for the input.